src/firepad-monaco.ts

```diff
@@ -29,31 +27,67 @@ export function fromMonaco(
   const userName: string = options.userName || userId.toString();
   const defaultText: string = options.defaultText || editor.getValue();
 
-  let databaseAdapter: IDatabaseAdapter;
-
-  if (firestoreRef) {
-    databaseAdapter = new FirestoreAdapter(
-      databaseRef,
-      firestoreRef,
-      userId,
-      userColor,
-      userName
-    );
-  } else {
-    databaseAdapter = new FirebaseAdapter(
-      databaseRef,
-      userId,
-      userColor,
-      userName
-    );
-  }
+  let databaseAdapter: IDatabaseAdapter = new FirebaseAdapter(
+    databaseRef,
+    userId,
+    userColor,
+    userName
+  );
 
   const editorAdapter = new MonacoAdapter(editor, false);
+  return new Firepad(databaseAdapter, editorAdapter, {
+    userId,
+    userName,
+    userColor,
+    defaultText,
+  });
+}
+
+/**
+ * Creates a modern Firepad from Monaco editor.
+ * @param databaseRef - Firestore database document Reference.
+ * @param editor - Monaco Editor instance.
+ * @param options - Firepad constructor options (optional).
+ */
+export function fromMonacoWithFirestore(
+  databaseRef: firebase.firestore.DocumentReference, //TODO should we support path : string
+  editor: monaco.editor.IStandaloneCodeEditor,
+  options: Partial<IFirepadConstructorOptions> = {}
+): IFirepad {
+  // Initialize constructor options with their default values
+  const userId: UserIDType = options.userId || uuid();
+  const userColor: string =
+    options.userColor || Utils.colorFromUserId(userId.toString());
+  const userName: string = options.userName || userId.toString();
+  const defaultText: string = options.defaultText || editor.getValue();
+
+  let databaseAdapter: IDatabaseAdapter = new FirestoreAdapter(
+    null,
```

Comment:
the RTDB `databaseRef ` in first param can be removed since the adapter should solely work with firestore

Reply:
removed the rtdb ref